src/python/WMCore/MicroService/MSUnmerged/MSUnmerged.py

```diff
@@ -57,27 +70,144 @@ def execute(self):
+            except Exception as ex:
+                msg = "%s: General error from pipeline. RSE: %s. Error:  \n%s. "
+                msg += "\nWill retry again in the next cycle."
+                self.logger.exception(msg, self.plineUnmerged.name, rse['rse'], str(ex))
+                continue
+        return totalNumRses, totalNumFiles, numRsesCleaned, numFilesDeleted
+
+    def filterRseList(self, rseList):
+        """
+        Applies any additional constraints about the subset of RSEs on which
+        the current service instance is about to work
+        :param rseList: Initial list of RSEs from which the final set should be build
+        :return:        A list of RSEs
+        """
+        return rseList
+
+    def cleanFiles(self, rse):
+        """
+        The method to implement the actual deletion of files for an RSE.
+        :param rse: MSUnmergedRSE object to be cleaned
+        :return:    The MSUnmergedRSE object
+        """
+        try:
+            # for fileUnmerged in rse['files']['toDelete']:
+            #     try:
+            #         self.gfalCommand(rse['delInterface'], fileUnmerged)
+            #         rse['counters']['numFilesDeleted'] += 1
+            #         rse['files']['deletedSuccess'].append(fileUnmerged)
+            #     except Exception as ex:
+            #         rse['files']['deletedFail'].append(fileUnmerged)
+            #         msg = "Error while trying to delete file: %s for RSE: %s"
+            #         msg += "Will retry in the next cycle. Err: %s"
+            #         self.logger.debug(msg, fileUnmerged, rse['name'], str(ex))
+            rse['isClean'] = self._checkClean(rse)
+        except Exception as ex:
+            msg = "Error while cleaning RSE: %s"
+            msg += "Will retry in the next cycle. Err: %s"
+            self.logger.debug(msg, rse['name'], str(ex))
+
+        return rse
+
+    def _checkClean(self, rse):
+        """
+        A simple function to check if every file in an RSE's unmerged area have
+        been deleted
+        :param rse: The RSE to be checked
+        :return:    Bool: True if all files found have been deleted, False otherwise
+        """
+        return rse['counters']['totalNumFiles'] == rse['counters']['numFilesDeleted']
+
+    def resetCounters(self):
+        """
+        A simple function for zeroing the deleted files counters.
+        """
+
+        for rse in self.rseCounters:
+            self.rseCounters[rse]['deletedSuccess'] = 0
+            self.rseCounters[rse]['deletedFail'] = 0
+
+        for pline in self.plineCounters:
+            self.plineCounters[pline.name]['deletedSuccess'] = 0
+            self.plineCounters[pline.name]['deletedFail'] = 0
+
     def getRSEList(self):
         """
-        Queries CRIC for the proper RSE lists to iterate through.
-        return: List of RSEs.
+        Queries Rucio for the proper RSE list to iterate through.
+        :return: List of MSUnmergedRSE objects.
```

Comment:
Which is exactly correct. 
This was  a docstring reflecting the return from this method  in a previous version when I was building the list of `MSUnmergedRSE` objects directly here. but I left that behind - because it means that   once we  start accumulating  all those huge lists of deletions and information in the objects we will have to keep them until the exit. I'd rather prefer to generate a simple list of identifiers (RSE names) here and pass it up and generate those objects right before I through them in the pipeline - this way, once the `MSUnmergedRSE` object goes through, the only leftover should be the counters.... 

Thanks for catching this docstring inconsistency. I will fix it.

Reply:
Fixed